src/util/compat.hpp

```diff
@@ -27,8 +27,14 @@ constexpr bool using_gnu_compiler(int major=0, int minor=0, int patchlevel=0) {
 
 // std::end() broken with (at least) xlC 13.1.4.
 
+namespace impl {
+    using std::end;
```

Comment:
I don't feel comfortable putting `std::end `in the `impl` namespace. How about renaming `impl::end_`, `impl::end`, and explicitly using `std::end`?

Reply:
This is all for correct ADL of `end()` — there are similar cases scattered elsewhere through the code. It should only be a problem if we try and use `arb::impl::end` (or `begin` etc.) for something else. We could use a different inner namespace than `impl`, but I'm inclined to leave it as is until or unless we bump into this sort of issue.